### DISCUSSION TITLE
How to convert Pandera Schemas to Snowflake Schemas?

### DISCUSSION THREAD

**Discussion by nehiljain** [recap-build/recap#205] [Q&A] [answered]
Hi,

I have a bunch of [pandera schemas](https://pandera.readthedocs.io/en/stable/dataframe_schemas.html). I am using [Dagster output validation with Pandera](https://docs.dagster.io/integrations/pandera). I also want to convert this schema to snowflake to check for column names and types in snowflake tables that map 1:1 with these schemas.

One idea that comes to mind is:

Write a pandera convert to recap schema.
Write a __eq__ method for recap schema.
Using sqlalchemy inspect create a recap schema from the specific snowflake table and check if the two schemas are equal. 

![image](https://user-images.githubusercontent.com/2761999/222832110-560c4366-bb3d-4b6d-901d-e36e3174274e.png)
Thinking about it in this sequence.

**Comment by criccomini** [comment, marked as answer]
You're correct. You'd need to write a to_schema method that converts pandera schemas to Recap schemas. From there, you can do all the normal stuff with Recsp schemas like converting it to JSON schema or table DDLs.

Recap supports naive equality right now (just comparing Pydantic model fields), so `==` should work provided you want those semantics. For a looser sense of equality, we'll need to implement a compatible() method.
